File: leetcode/0014-longest-common-prefix/solution.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        a = min(strs)
        b = max(strs)

        i = 0
        while i < len(a) and a[i] == b[i]:
            i += 1
        
        return a[:i]

File: leetcode/0014-longest-common-prefix/solution.rs
impl Solution {
    pub fn longest_common_prefix(strs: Vec<String>) -> String {
        let mut prefix = strs[0].as_bytes();

        for s in strs.iter().skip(1) {
            let n = prefix
                .iter()
                .zip(s.as_bytes())
                .take_while(|(a, b)| a == b)
                .count();
            
            prefix = &prefix[..n];

            if prefix.is_empty() {
                return String::new();
            }
        }

        String::from_utf8(prefix.to_vec()).unwrap()
    }
}
